lms/djangoapps/grades/tests/test_new.py

```diff
@@ -239,56 +240,155 @@ def test_save_and_load(self):
+    def load_scoreable_course(cls):
+        """
+        This test course lives at common/test/data/scoreable.
+
+        It contains six scoreable blocks in one sequence:
+
+        - CAPA
+        - ORA
+        - SGA (Staff Graded Assignment)
+        - LTI
+        - LTI Consumer
+        - Randomized Content Block (containing Library Content with CAPA
+          problems)
+
+        To expand coverage to other block types, you can either edit the course
+        xml directly, or do the following:
+
+        1. Zip up the scoreable directory into a tarball:
+
+            tar cvzf course.tar.gz common/test/data/scoreable'
+
+        2. Import the tarball into studio.
+        3. Add the new blocks.
+        4. Export the modified course.
+        5. Unzip to a temporary directory:
+
+            cd /tmp
+            tar xvzf ~/Downloads/course.*.tar.gz
+            somewhere, and
+
+        6. Copy the data back into the test directory:
+
+            rsync -avz --delete-after /tmp/course/ /path/to/edx-platform/common/test/data/scoreable
+
+        You will need to update SCORED_BLOCK_COUNT and ACTUAL_TOTAL_POSSIBLE
+        above to reflect the real number of scoreable blocks in the course.
+        """
+        course_items = import_course_from_xml(
+            cls.store,
+            'test_user',
+            TEST_DATA_DIR,
+            source_dirs=['scoreable'],
+            static_content_store=None,
+            target_id=cls.store.make_course_key('edX', 'scoreable', '3000'),
+            raise_on_failure=True,
+            create_if_not_present=True,
+        )
+
+        cls.course = course_items[0]
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMultipleProblemTypesSubsectionScores, cls).setUpClass()
+        cls.load_scoreable_course()
+        chapter1 = cls.course.get_children()[0]
+        cls.seq1 = chapter1.get_children()[0]
+
+    def test_score_submission_for_all_problems(self):
+        course_structure = get_course_blocks(self.student, self.course.location)
+        subsection_factory = SubsectionGradeFactory(
+            self.student,
+            course_structure=course_structure,
+            course=self.course,
         )
+        score = subsection_factory.update(self.seq1)
+
+        self.assertEqual(score.all_total.earned, 0.0)
+        self.assertEqual(score.all_total.possible, self.ACTUAL_TOTAL_POSSIBLE)
+
+        # The actual possible score for the six provided blocks.  The number
+        # below is from mocking.
+
+        earned_per_block = 3.0
+        possible_per_block = 7.0
+        with mock_get_score(earned_per_block, possible_per_block):
+            score = subsection_factory.update(self.seq1)
+        self.assertEqual(score.all_total.earned, self.SCORED_BLOCK_COUNT * earned_per_block)
+        self.assertEqual(score.all_total.possible, self.SCORED_BLOCK_COUNT * possible_per_block)
+
+
+@ddt.ddt
+class TestVariedMetadata(ModuleStoreTestCase, ProblemSubmissionTestMixin):
```

Comment:
Declare the concrete class at the end of the class list - otherwise, it results in unexpected behavior (like overridden methods in the mixin won't get called).

_"Mixins come first even though it looks odd."_ from Ned's: http://nedbatchelder.com/blog/201210/multiple_inheritance_is_hard.html.